### DISCUSSION TITLE
Guide to BLAST databases?

### DISCUSSION THREAD

**Discussion by aspencoyle** [RobertsLab/resources#1130] [Q&A] [answered]
I'm looking to BLAST a transcriptome (specifically cbai_transcriptomev2.0) against the whole NCBI database with a taxonomy filter to determine which sequences are _Hematodinium_ and which are _C. bairdi_.

In the mox folder /gscratch/srlab/blastdbs/, there's quite a few existing databases, as follows:

UniProtKB_20181008

UniProtKB_20190109

ncbi-nr-20190925

ncbi-nr-20200924

ncbi-nr-nt-20181114

ncbi-nr-nt-v5

ncbi-sp-v5

ncbi-sp-v5_20210224

uniprot_sprot_20200123

I assume I want to use either one of the ncbi-nr or ncbi-sp databases, but I really have no clue which I should use! Is there some kind of guide defining each database?

**Comment by kubu4** [comment]
Do the readme files in those folders offer any insight?

**Reply by aspencoyle** [reply]
Sadly there is no readme file in the folder!

**Comment by kubu4** [comment]
Which folder? You listed nine folders above.

**Reply by aspencoyle** [reply]
...well dang, that solves that - I assumed each of those were files, just didn't notice the lack of an extension! I'll check out the readme files and see if that solves my question

**Comment by kubu4** [comment]
Also, I'd probably lean towards the most recent:

`ncbi-sp-v5_20210224`

**Comment by aspencoyle** [comment]
Alright, after consulting the README files, these seem like the good candidates:

# ncbi-nr-20190925: 

Directory containing files for BLASTing against the NCBI nr database.

nr FastA file was downloaded 20200924 by SJW from:

ftp://ftp.ncbi.nih.gov/blast/db/FASTA/nr.gz


---

FILES

- *.dmp*: NCBI Taxonomy database dump files. Files were downloaded 20200924 by SJW from:

ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdmp.zip

- nr.dmnd: DIAMOND "BlAST" database. Generated 20200924 by SJW with the following command:

/gscratch/srlab/programs/diamond-2.0.4/diamond makedb --in nr.faa -d nr --taxonmap prot.accession2taxid.gz --taxonnodes nodes.dmp --taxonnames names.dmp

- nr.faa: NCBI nr protein FastA file.

- prot.accession2taxid.gz: NCBI protein accession number to taxon IDs mapping file. Downloaded 20200924 by SJW from:

ftp://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/prot.accession2taxid.gz

# ncbi-nr-nt-v5

NCBI non-redundant (nr) nucleotide (nt) database v5.

Downloaded 20190105 from:

https://ftp.ncbi.nlm.nih.gov/blast/db/v5/

# ncbi-sp-v5_20210224

/gscratch/srlab/blastdbs/ncbi-sp-v5_20210224

NCBI SwissProt V5 database. Specifically for use with NCBI BLAST 2.8.1+ (and higher).

Downloaded 200210224 by SJW from:

https://ftp.ncbi.nlm.nih.gov/blast/db/v5/swissprot_v5.tar.gz

# End of choices

Alright, so of the three, I'm not quite sure which would be optimal. Would it be optimal to determine taxa by compare the transcriptome to nucleotide sequences, or should I instead use BLASTx to compare to protein sequences? And if the latter, which database should be used?

**Comment by aspencoyle** [comment, marked as answer]
Answered today during class - I'll be using ncbi-sp-v5_20210224!

**Comment by sr320** [comment]
for documentation purposes note that ncbi has nothing to do with
swiss-prot, thus labelling is confusing.

**Comment by kubu4** [comment]
Sort of. It's a SwissProt BLAST library supplied by NCBI.

**Reply by sr320** [reply]
I see now.. thanks for clarification

**Comment by aspencoyle** [comment]
Hmm, alright, gotcha. So my goal is to BLAST Transcriptome v2.0 against the entire NCBI database - in that case, should I choose to BLAST against ncbi-nr-20190925 (most recent, unsure if a standard BLAST is possible, given that the README indicates it's a DIAMOND BLAST database) or the NCBI non-redundant nucleotide database (downloaded 2019-01-05)? Or does the Swiss-Prot database still fit the purposes here?
